app/models/user.rb

```diff
@@ -1,8 +1,11 @@
 class User < ApplicationRecord
-  has_many :comments, foreign_key: :AuthorId
-  has_many :posts, foreign_key: :AuthorId
-  has_many :likes, foreign_key: :AuthorId
+  has_many :comments, foreign_key: :author_id
+  has_many :posts, foreign_key: :author_id
+  has_many :likes, foreign_key: :author_id
 
+  validates :name, presence: true, length: { minimum: 3, maximum: 20 }
+  validates :bio, presence: true, length: { minimum: 3, maximum: 150 }
+  validates :posts_counter, presence: true, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
```

Comment:
- [ ] Kindly make sure the validation for the `posts_counter` column is of type `integer` and not `numericality` as required for the project. If you check project the requirements you will find that it is mentioned as follow: 
**PostsCounter must be an integer greater than or equal to zero.**

Reply:
Kindly refer to this guide you will find the syntax i wrote is correct [link section 2.9](https://guides.rubyonrails.org/active_record_validations.html#validations-overview)

Reply:
Hi @mohamed20163858 

You are true